azkaban-common/src/main/java/azkaban/jobcallback/JobCallbackManager.java

```diff
@@ -166,76 +172,79 @@ private void processJobCallOnFinish(final Event event) {
         JobCallbackUtil.parseJobCallbackProperties(props, COMPLETED,
-            contextInfo, maxNumCallBack, this.logger);
+            contextInfo, maxNumCallBack, logger);
 
     // now make the call
     if (!httpRequestsForCompletedStatus.isEmpty()) {
-      this.logger.info("Making " + httpRequestsForCompletedStatus.size()
+      jobLogger.info("Making " + httpRequestsForCompletedStatus.size()
           + " job callbacks for status: " + COMPLETED);
 
       addDefaultHeaders(httpRequestsForCompletedStatus);
-      JobCallbackRequestMaker.getInstance().makeHttpRequest(jobId, this.logger,
+      JobCallbackRequestMaker.getInstance().makeHttpRequest(jobId, logger,
           httpRequestsForCompletedStatus);
     } else {
-      this.logger.info("No job callbacks for status: " + COMPLETED);
+      logger.info("No job callbacks for status: " + COMPLETED);
     }
   }
 
-  private void processJobCallOnStart(final Event event) {
-    final JobRunner jobRunner = (JobRunner) event.getRunner();
-
-    if (JobCallbackUtil.isThereJobCallbackProperty(jobRunner.getProps(),
+  private void processJobCallOnStart(final Event event, final Props jobProps,
+      final Logger jobLogger) {
+    if (JobCallbackUtil.isThereJobCallbackProperty(jobProps,
         JobCallbackStatusEnum.STARTED)) {
 
       // don't want to waste time resolving properties if there are
       // callback properties to parse
-      final Props props = PropsUtils.resolveProps(jobRunner.getProps());
+      // resolve props in best effort. Ignore the case where a referenced property is undefined.
+      final Props props = PropsUtils.resolveProps(jobProps, true);
 
       final Map<String, String> contextInfo =
           JobCallbackUtil.buildJobContextInfoMap(event, this.azkabanHostName);
 
       final List<HttpRequestBase> jobCallbackHttpRequests =
           JobCallbackUtil.parseJobCallbackProperties(props, STARTED,
-              contextInfo, maxNumCallBack, this.logger);
+              contextInfo, maxNumCallBack, logger);
 
       final String jobId = contextInfo.get(CONTEXT_JOB_TOKEN);
       final String msg =
           String.format("Making %d job callbacks for job %s for jobStatus: %s",
               jobCallbackHttpRequests.size(), jobId, STARTED.name());
 
-      jobRunner.getLogger().info(msg);
+      jobLogger.info(msg);
 
       addDefaultHeaders(jobCallbackHttpRequests);
 
       JobCallbackRequestMaker.getInstance().makeHttpRequest(jobId,
-          this.logger, jobCallbackHttpRequests);
+          logger, jobCallbackHttpRequests);
     }
   }
 
   private String getAzkabanHostName(final Props props) {
-    final String baseURL = props.get(JobRunner.AZKABAN_WEBSERVER_URL);
+    final String baseURL = props.get(AZKABAN_WEBSERVER_URL);
     try {
-      String hostName = InetAddress.getLocalHost().getHostName();
+      String hostName =
+          props.getString("jetty.hostname", "localhost") + ":" + (props.getBoolean(JETTY_USE_SSL
```

Comment:
Can you use constants instead of strings?

Reply:
Good point. Done.